src/pyconverter/xml2py/writer.py

```diff
@@ -295,90 +391,246 @@ def write_source(
+        #             subprocess.run(["python", str(file_path)])
+        #         except:
+        #             print(python_method)
+        #             raise RuntimeError(f"Failed to execute {file_path}") from None
+
+        #         class_structure[file_name] = [class_name, methods_structure]
+        #     package_structure[module_name] = class_structure
+
+    for command_name in name_map.keys():
+        if command_name not in all_commands:
+            print(f"{command_name} is not in the structure map")
+
+    write_global__init__file(library_path, module_name_list)
+    write__init__file(library_path)
+
+    print(f"Commands written to {library_path}")
 
     # copy package files to the package directory
-    copy_package(template_path, new_package_path, clean)
+    copy_template_package(template_path, new_package_path, clean)
     graph_path = get_paths(xml_doc_path)[0]
     shutil.copytree(graph_path, os.path.join(new_package_path, "doc", "source", "images"))
-
-    return cmd_path
+    return package_structure
 
 
-def write_docs(commands, cmd_map, package_path):
+def write_docs(package_path, package_structure=None, config_path="config.yaml"):
     """Output to the autogenerated ``package`` directory.
 
     Parameters
     ----------
-    commands : list[XMLCommand]
-        List of XML commands.
+    name_map : dict
+        Dictionary with the following format: ``{"initial_command_name": "python_name"}``.
 
-    cmd_map : dict
-        Dictionary with this format: ``{"command_name": "python_name"}``.
-
-    path : str
+    package_path : str
         Path to the new package folder.
 
+    module_name_list : list
+        List of module names created.
+
+    package_structure :
+        Dictionary with the following format:
+        ``{'python_module_name': [{'python_class_name': python_names_list}]}``.
+
     Returns
     -------
     str
         Path to the new document page.
 
     """
+    library_name = get_config_data_value(config_path, "library_name_structured")
+    if library_name[0] == "src":
+        library_name.pop(0)
+    library_name = ".".join(library_name)
 
     doc_package_path = os.path.join(package_path, "doc/source")
     if not os.path.isdir(doc_package_path):
         os.makedirs(doc_package_path)
 
     doc_src = os.path.join(doc_package_path, "docs.rst")
     with open(doc_src, "w") as fid:
-        fid.write("Autosummary\n")
-        fid.write("===========\n\n")
-
-        fid.write(".. currentmodule:: pyconverter.generatedcommands\n\n")
-        fid.write(".. autosummary::\n")
-        fid.write("   :template: base.rst\n")
-        fid.write("   :toctree: _autosummary/\n\n")
-        for ans_name in commands:
-            if ans_name not in SKIP_XML:
-                cmd_name = cmd_map[ans_name]
-                fid.write(f"   {cmd_name}\n")
+        fid.write(
+            """
+API documentation
+==================
+
+.. toctree::
+   :maxdepth: 2
+   :hidden:\n\n
+"""
+        )
```

Comment:
I like this more than the previous repeated ``fid.write``.

Furthermore, for testing purposes, you might want to generate the full string first, and separate the content generation from the file writing, so eventually you could tests the content, without writing the file.